src/webextension/manage/containers/modal-root.js

```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { withLocalization } from "fluent-react";
+import PropTypes from "prop-types";
+import React from "react";
+import Modal from "react-modal";
+import { connect } from "react-redux";
+
+import CancelEditingModal from "./modals/cancel-editing";
+import DeleteItemModal from "./modals/delete-item";
+
+import styles from "./modal-root.css";
+
+const MODALS = {
+  "cancel": CancelEditingModal,
+  "delete": DeleteItemModal,
+};
+
+function ModalRoot({modal, getString}) {
+  if (!modal) {
+    return null;
+  }
+  const CurrentModal = MODALS[modal];
+
+  // For mysterious reasons, <Localization/> will break <Modal/>, so we
+  // use getString() instead.
+  return (
+    <Modal isOpen={true} contentLabel={getString("modal-root-title")}
+           className={styles.modal} overlayClassName={styles.overlay}>
+      <CurrentModal/>
+    </Modal>
+  );
+}
+
+ModalRoot.propTypes = {
+  modal: PropTypes.string,
+  getString: PropTypes.func.isRequired,
+};
+
+export default connect(
+  (state) => ({
+    modal: state.modal.modal,
```

Comment:
I would consider `.name` instead of `.modal`.  Where it's used it should be clear which one is the target.

Reply:
Good point. I went with `.id`, since I think that emphasizes that it's a unique internal name.